### ISSUE TITLE
Error rendering Maven report: Unable to determine output directory

### ISSUE THREAD

**Issue by benze** [dependency-check/DependencyCheck#193]
I have yet to truly figure out a way to build a sample project to show this error, but I get the following error out of Maven when I try to use the aggregate report in 1.2.8:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-site-plugin:3.3:site (default-site) on project ComPublic: Error during page generation: Error rendering Maven report: Unable to determine output directory for 'Maven Distribution Common Module' -> [Help 1]
```

I enabled the maven debugging  (-X) and get the following stack trace.  I am hoping that it can trigger an idea of what may be causing this.

```
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-site-plugin:3.3:site (default-site) on project ComPublic: Error during page generation
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:216)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:120)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:347)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:154)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:582)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:214)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:158)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:88)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
        at java.lang.reflect.Method.invoke(Method.java:618)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Error during page generation
        at org.apache.maven.plugins.site.SiteMojo.execute(SiteMojo.java:143)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:132)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
        ... 19 more
Caused by: org.apache.maven.doxia.siterenderer.RendererException: Error rendering Maven report: Unable to determine output directory for 'Maven Distribution Common Module'
        at org.apache.maven.plugins.site.ReportDocumentRenderer.renderDocument(ReportDocumentRenderer.java:233)
        at org.apache.maven.doxia.siterenderer.DefaultSiteRenderer.renderModule(DefaultSiteRenderer.java:319)
        at org.apache.maven.doxia.siterenderer.DefaultSiteRenderer.render(DefaultSiteRenderer.java:135)
        at org.apache.maven.plugins.site.SiteMojo.renderLocale(SiteMojo.java:175)
        at org.apache.maven.plugins.site.SiteMojo.execute(SiteMojo.java:138)
        ... 21 more
Caused by: org.apache.maven.reporting.MavenReportException: Unable to determine output directory for 'Maven Distribution Common Module'
        at org.owasp.dependencycheck.maven.BaseDependencyCheckMojo.generate(BaseDependencyCheckMojo.java:387)
        at org.owasp.dependencycheck.maven.BaseDependencyCheckMojo.generate(BaseDependencyCheckMojo.java:367)
        at org.apache.maven.plugins.site.ReportDocumentRenderer.renderDocument(ReportDocumentRenderer.java:228)
        ... 25 more
Caused by: org.apache.maven.plugin.MojoExecutionException: Unable to determine output directory for 'Maven Distribution Common Module'
        at org.owasp.dependencycheck.maven.BaseDependencyCheckMojo.getCorrectOutputDirectory(BaseDependencyCheckMojo.java:415)
        at org.owasp.dependencycheck.maven.AggregateMojo.runCheck(AggregateMojo.java:117)
        at org.owasp.dependencycheck.maven.BaseDependencyCheckMojo.generate(BaseDependencyCheckMojo.java:385)
        ... 27 more
```

pom.xml:

```
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
  <modelVersion>4.0.0</modelVersion>
  <groupId>com.bic</groupId>
  <artifactId>MavenDistributionCommon</artifactId>
  <packaging>pom</packaging>
  <version>1.4.2</version>
  <name>Maven Distribution Common Module</name>

    <distributionManagement>
        <repository>
            <id>${releases.server.id}</id>
            <name> Releases</name>
            <url>${release.repo.url}</url>
        </repository>
        <snapshotRepository>
            <id>${snapshots.server.id}</id>
            <name> Snapshots</name>
            <url>${snapshots.repo.url}</url>
        </snapshotRepository>
    </distributionManagement>

</project>
```

This pom is the parent to my orchestrator pom, however, it doesn't seem to make a difference if I remove it or not as the parent.  Originally I thought that the lack of dependency defn in this pom was causing the error, but I added in some dummy deps and it made no difference.

**Comment by benze** [comment]
Overnight I had a sudden inspiration.  Due to the way I have the poms structured, the modules actually now were inheriting the orchestrator pom (and hence inheriting the <reporting> section).  If I break that inheritance, then the mvn site builds to completion properly.

I tried adding `inherited` to false in my reportSet defn but it doesn't seem to make any difference.

```
<reporting>
        <outputDirectory>target/site</outputDirectory>
        <plugins>
            <plugin>
                <groupId>org.owasp</groupId>
                <artifactId>dependency-check-maven</artifactId>
                <version>1.2.8</version>
                <reportSets>
                    <reportSet>
                        <inherited>false</inherited>
                        <reports>
                            <report>aggregate</report>
                        </reports>
                    </reportSet>
                </reportSets>
            </plugin>
        </plugins>
    </reporting>
```

**Comment by jeremylong** [comment]
I'm going to ask a dumb question - if you don't include the dependency-check plugin, rather replace it with something like the PMD Plugin, do you get the same error?

``` xml
<plugin>
      <artifactId>maven-pmd-plugin</artifactId>
      <version>3.0.1</version>
        <reportSet>
          <!-- defines pmd aggregate report -->
          <id>pmd-aggregate</id>
          <inherited>false</inherited>
          <reports>
            <report>pmd</report>
          </reports>
          <configuration>
            <aggregate>true</aggregate>
          </configuration>
        </reportSet>
      </reportSets>
    </plugin>
```

**Comment by benze** [comment]
A colleague at work ran into the same issue on a different project and spent some time putting together a repeatable case.  These were his findings:

The root of the problem seems to be if the aggregator pom has a parent pom which contains dependencies. 
-   aggregator: pom for the aggregation of modules parent1, project1, project2 and project3.
-   parent1: parent of project1, project2 and project3. Contains a dependency.
-   parent2: by default, no one’s parent. Contains a dependency.
-   parent3: by default, no one’s parent. Does not contain a dependency.
-   project1, project 2, project 3: simple projects with 1 dependency each.

Scenarios:
1.  If aggregator does not have a parent or has a parent with no dependencies (parent3), the build goes through correctly.
2.  If aggregator has a parent with a dependency (parent1 which also is the parent of the other sub-projects and is included in the aggregator pom, or parent2 which is otherwise no one else’s parent), then the build will fail with the following error.

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] parent1 ........................................... SUCCESS [0.612s]
[INFO] project1 .......................................... SUCCESS [0.863s]
[INFO] project2 .......................................... SUCCESS [0.439s]
[INFO] project3 .......................................... SUCCESS [0.489s]
[INFO] aggregate ......................................... FAILURE [9.248s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 11.757s
[INFO] Finished at: Tue Jan 27 15:16:55 EST 2015
[INFO] Final Memory: 38M/512M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-site-plugin:3.3:site (default-site) on project aggregate: Error during page generation: Error rendering Maven report: Unable to determine output directory for 'parent1' -> [Help 1]
```

So the solutions are that either the aggregator pom must not have a parent pom, or that parent pom must not have dependencies.

I have a zip to represent the above issue, but not sure how to attach it to this issue.

**Comment by jeremylong** [comment]
A patch has been added to 1.2.9-SNAPSHOT to resolve this issue. More regression testing needs to be done prior to closing this issue.

**Comment by jeremylong** [comment]
Further testing by the reporter was completed and confirms that this issue is closed in 1.2.9-SNAPSHOT.
